src/views/catalog/customize/components/DiskSource/DiskSource.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import {
+  Checkbox,
+  FormGroup,
+  Popover,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextInput,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+
+import { PersistentVolumeClaimSelect } from '../PersistentVolumeClaimSelect';
+
+import {
+  DISK_SOURCE,
+  HTTP_DISK_SOURCE_NAME,
+  PVC_DISK_SOURCE_NAME,
+  REGISTRY_DISK_SOURCE_NAME,
+} from './constants';
+import { getGenericDiskSourceCustomization, getPVCDiskSource } from './utils';
+import { VolumeSize } from './VolumeSize';
+import { DISK_SOURCE_OPTIONS_IDS, PVC_SIZE_FORMATS } from '.';
+
+import './DiskSource.scss';
+
+type DiskSourceProps = {
+  onChange: (diskSource: DISK_SOURCE) => void;
+};
+
+export const DiskSource: React.FC<DiskSourceProps> = ({ onChange }) => {
+  const { t } = useKubevirtTranslation();
+
+  const [selectedSourceType, setSourceType] =
+    React.useState<DISK_SOURCE_OPTIONS_IDS>(PVC_DISK_SOURCE_NAME);
+  const [pvcNameSelected, selectPVCName] = React.useState<string>();
+  const [pvcNamespaceSelected, selectPVCNamespace] = React.useState<string>();
+  const [volumeSize, setVolumeSize] = React.useState(30);
+  const [volumeFormat, setVolumeFormat] = React.useState(PVC_SIZE_FORMATS.GiB);
+  const [httpURL, setHTTPURL] = React.useState('');
+  const [containerImage, setContainerImage] = React.useState('');
+
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const onSelectDiskSource = React.useCallback((event, selection) => {
+    setSourceType(selection);
+    setIsOpen(false);
+  }, []);
+
+  const diskSourceLabel = (
+    <>
+      {t('Disk source')}{' '}
+      <Popover
+        aria-label={t('Help')}
+        bodyContent={() => (
+          <div>
+            {t(
+              'Disk Source represents the source for our Disk, this can be HTTP, Registry or an excisting PVC',
+            )}
+          </div>
+        )}
+      >
+        <HelpIcon />
+      </Popover>
+    </>
+  );
+
+  React.useEffect(() => {
```

Comment:
@yaacov please test this change locally and see if it catches all use cases if you can